### ISSUE TITLE
Error in regions export when using strict typescript compiler settings

### ISSUE THREAD

**Issue by breathe** [nozzlegear/typed-countries#1]
This export is broken when used in projects with strict typescript compiler settings ...

```
export const regions: Region[] = countries.reduce(
  (result, country) => {
    if (result.indexOf(country.region) === -1) {
      result.push(country.region);
    }

    return result;
  },
  []
);
```

Inferred type of [] passed to reduce is never[] ...

Fix looks like this:

```
export const regions: Region[] = countries.reduce(
  (result, country) => {
    if (result.indexOf(country.region) === -1) {
      result.push(country.region);
    }

    return result;
  },
  [] as Region[]
);
```

**Comment by nozzlegear** [comment]
Interesting, thanks for letting me know! I'll get a fix published for this as soon as I get a chance.




--


Josh Harms







On Mon, Jun 4, 2018 at 8:38 PM -0500, "breathe" <notifications@github.com> wrote:












This export is broken when used in projects with strict typescript compiler settings ...
export const regions: Region[] = countries.reduce(
  (result, country) => {
    if (result.indexOf(country.region) === -1) {
      result.push(country.region);
    }

    return result;
  },
  []
);
``

Inferred type of [] passed to reduce is never[] ...

Fix looks like this:




export const regions: Region[] = countries.reduce(

(result, country) => {

if (result.indexOf(country.region) === -1) {

result.push(country.region);

}
return result;



},

[] as Region[]

);

**Comment by henrinormak** [comment]
Also seeing this, would love to get rid of the error.

**Comment by nozzlegear** [comment]
Thanks for bumping this, it slipped my mind! I've published the fix in 1.1.1 on NPM, please let me know if it doesn't work.

**Comment by henrinormak** [comment]
Thanks for the super quick response, it works fine now!
